src/url-loader/package-url-resolver.ts

```diff
@@ -57,10 +57,12 @@ export class PackageUrlResolver extends UrlResolver {
   }
 
   resolve(
-      unresolvedHref: FileRelativeUrl|PackageRelativeUrl,
-      baseUrl: ResolvedUrl = this.packageUrl,
+      firstHref: ResolvedUrl|PackageRelativeUrl, secondHref?: FileRelativeUrl,
       _import?: ScannedImport): ResolvedUrl|undefined {
-    const resolvedHref = this.simpleUrlResolve(unresolvedHref, baseUrl);
+    const [baseUrl, unresolvedHref] = typeof secondHref === 'undefined' ?
```

Comment:
This logic is pretty universally used in url resolvers, and it is a bit subtle. What about putting it in a protected method on UrlResolver? `getBaseAndUnresolved`?

Reply:
I like that idea!  Added `getBaseAndUnresolved`.